### ISSUE TITLE
Problem: AIP restore sends AIPs to the wrong directory, and with the wrong name

### ISSUE THREAD

**Issue by jarrodharvey** [archivematica/Issues#420]
**Expected behaviour**
The AIP restore process should replace the corrupted AIP with the recovery copy.

**Current behaviour**
The recovery copy is saved in the parent directory, and has the same name as the parent directory.

For further information, please refer to this thread: https://groups.google.com/forum/#!topic/archivematica/J3Ci9lMpi-M

**Steps to reproduce**
Follow the [process for AIP recovery](https://www.archivematica.org/fr/docs/storage-service-0.13/recovery/).

**Your environment (version of Archivematica, OS version, etc)**
Version 1.8 of Archivematica, with CentOS 7.

---
**For Artefactual use:**
Please make sure these steps are taken before moving this issue from Review to Verified in Waffle:

- All PRs related to this issue are properly linked 👍
- All PRs related to this issue have been merged 👍
- Test plan for this issue has been implemented and passed 👍
- Documentation regarding this issue has been written and it has been added to the release notes, if needed 👍

**Comment by replaceafill** [comment]
I ran into this today in my dev environment with the `stable/0.18.x` branch of the Storage Service.

Just for reference these are the logs after clicking the `Approve` button in the `Package Recover Requests` page on a compressed AIP.

```
INFO      2022-03-22 11:04:08  locations.models.package:package:extract_file:1627:  Extracting file with: ['unar', '-force-overwrite', '-o', '/var/archivematica/storage_service/tmpqbqeuk_x', '/var/archivematica/storage_service/recover/pictures_compressed-6fd60c75-7899-4da1-808e-0c959c798052.7z'] to /var/archivematica/storage_service/tmpqbqeuk_x/pictures_compressed-6fd60c75-7899-4da1-808e-0c959c798052
DEBUG     2022-03-22 11:04:09  locations.models.space:space:move_to_storage_service:346:  TO: src: var/archivematica/sharedDirectory/www/AIPsStore/6fd6/0c75/7899/4da1/808e/0c95/9c79/8052/pictures_compressed-6fd60c75-7899-4da1-808e-0c959c798052.7z
DEBUG     2022-03-22 11:04:09  locations.models.space:space:move_to_storage_service:347:  TO: dst: var/archivematica/storage_service/recover/backup
DEBUG     2022-03-22 11:04:09  locations.models.space:space:move_to_storage_service:348:  TO: staging: /var/archivematica/storage_service
INFO      2022-03-22 11:04:09  locations.models.space:space:move_rsync:552:  Moving from /var/archivematica/sharedDirectory/www/AIPsStore/6fd6/0c75/7899/4da1/808e/0c95/9c79/8052/pictures_compressed-6fd60c75-7899-4da1-808e-0c959c798052.7z to /var/archivematica/storage_service/var/archivematica/storage_service/recover/backup
INFO      2022-03-22 11:04:09  locations.models.space:space:move_rsync:594:  rsync command: ['rsync', '-t', '-O', '--protect-args', '-vv', '--chmod=Fug+rw,o-rwx,Dug+rwx,o-rwx', '-r', '/var/archivematica/sharedDirectory/www/AIPsStore/6fd6/0c75/7899/4da1/808e/0c95/9c79/8052/pictures_compressed-6fd60c75-7899-4da1-808e-0c959c798052.7z', '/var/archivematica/storage_service/var/archivematica/storage_service/recover/backup']
DEBUG     2022-03-22 11:04:09  locations.models.space:space:move_from_storage_service:423:  FROM: src: var/archivematica/storage_service/recover/backup
DEBUG     2022-03-22 11:04:09  locations.models.space:space:move_from_storage_service:424:  FROM: dst: var/archivematica/storage_service/recover/backup
INFO      2022-03-22 11:04:09  locations.models.space:space:move_rsync:552:  Moving from /var/archivematica/storage_service/var/archivematica/storage_service/recover/backup to /var/archivematica/storage_service/recover/backup
DEBUG     2022-03-22 11:04:09  locations.models.space:space:move_rsync:566:  os.rename failed, trying with normalized paths
DEBUG     2022-03-22 11:04:09  locations.models.space:space:move_rsync:578:  os.rename failed, falling back to rsync. Source: /var/archivematica/storage_service/var/archivematica/storage_service/recover/backup; Destination: /var/archivematica/storage_service/recover/backup
INFO      2022-03-22 11:04:09  locations.models.space:space:move_rsync:594:  rsync command: ['rsync', '-t', '-O', '--protect-args', '-vv', '--chmod=Fug+rw,o-rwx,Dug+rwx,o-rwx', '-r', '/var/archivematica/storage_service/var/archivematica/storage_service/recover/backup', '/var/archivematica/storage_service/recover/backup']
DEBUG     2022-03-22 11:04:09  locations.models.space:space:move_to_storage_service:346:  TO: src: var/archivematica/storage_service/recover/pictures_compressed-6fd60c75-7899-4da1-808e-0c959c798052.7z
DEBUG     2022-03-22 11:04:09  locations.models.space:space:move_to_storage_service:347:  TO: dst: var/archivematica/sharedDirectory/www/AIPsStore/6fd6/0c75/7899/4da1/808e/0c95/9c79/8052
DEBUG     2022-03-22 11:04:09  locations.models.space:space:move_to_storage_service:348:  TO: staging: /var/archivematica/storage_service
INFO      2022-03-22 11:04:09  locations.models.space:space:move_rsync:552:  Moving from /var/archivematica/storage_service/recover/pictures_compressed-6fd60c75-7899-4da1-808e-0c959c798052.7z to /var/archivematica/storage_service/var/archivematica/sharedDirectory/www/AIPsStore/6fd6/0c75/7899/4da1/808e/0c95/9c79/8052
INFO      2022-03-22 11:04:09  locations.models.space:space:move_rsync:594:  rsync command: ['rsync', '-t', '-O', '--protect-args', '-vv', '--chmod=Fug+rw,o-rwx,Dug+rwx,o-rwx', '-r', '/var/archivematica/storage_service/recover/pictures_compressed-6fd60c75-7899-4da1-808e-0c959c798052.7z', '/var/archivematica/storage_service/var/archivematica/sharedDirectory/www/AIPsStore/6fd6/0c75/7899/4da1/808e/0c95/9c79/8052']
DEBUG     2022-03-22 11:04:09  locations.models.space:space:move_from_storage_service:423:  FROM: src: var/archivematica/sharedDirectory/www/AIPsStore/6fd6/0c75/7899/4da1/808e/0c95/9c79/8052
DEBUG     2022-03-22 11:04:09  locations.models.space:space:move_from_storage_service:424:  FROM: dst: var/archivematica/sharedDirectory/www/AIPsStore/6fd6/0c75/7899/4da1/808e/0c95/9c79/8052
INFO      2022-03-22 11:04:09  locations.models.space:space:move_rsync:552:  Moving from /var/archivematica/storage_service/var/archivematica/sharedDirectory/www/AIPsStore/6fd6/0c75/7899/4da1/808e/0c95/9c79/8052 to /var/archivematica/sharedDirectory/www/AIPsStore/6fd6/0c75/7899/4da1/808e/0c95/9c79/8052
DEBUG     2022-03-22 11:04:09  locations.models.space:space:move_rsync:566:  os.rename failed, trying with normalized paths
DEBUG     2022-03-22 11:04:09  locations.models.space:space:move_rsync:578:  os.rename failed, falling back to rsync. Source: /var/archivematica/storage_service/var/archivematica/sharedDirectory/www/AIPsStore/6fd6/0c75/7899/4da1/808e/0c95/9c79/8052; Destination: /var/archivematica/sharedDirectory/www/AIPsStore/6fd6/0c75/7899/4da1/808e/0c95/9c79/8052
INFO      2022-03-22 11:04:09  locations.models.space:space:move_rsync:594:  rsync command: ['rsync', '-t', '-O', '--protect-args', '-vv', '--chmod=Fug+rw,o-rwx,Dug+rwx,o-rwx', '-r', '/var/archivematica/storage_service/var/archivematica/sharedDirectory/www/AIPsStore/6fd6/0c75/7899/4da1/808e/0c95/9c79/8052', '/var/archivematica/sharedDirectory/www/AIPsStore/6fd6/0c75/7899/4da1/808e/0c95/9c79/8052']
```

**Comment by JBosse-Artefactual** [comment]
The issue has been resolved in 1.17. The [recovery process](https://www.archivematica.org/fr/docs/storage-service-0.22/recovery/) can now replace both compressed and uncompressed AIPs as described in the documentation. Ensure the AIP recovery directory is empty before starting the recovery process, and if a ``recover`` directory exists inside the internal processing location (e.g. ``/var/archivematica/storage_service/var/archivematica/storage_service/recover``), make sure it is empty as well.
